lib/updaters/types/pom.js

```diff
@@ -0,0 +1,25 @@
+const detectIndent = require('detect-indent')
+
+function versionTagRegex (contents) {
+  //  The version is in an xml tag with a single indent (remember, there are
+  //  normally many other version tags, e.g. for dependencies).
+  const { indent } = detectIndent(contents)
+  return RegExp(`^${indent}<version>([\\d\\.]+)<\\/version>`, 'm')
+}
+
+module.exports.readVersion = function (contents) {
+  const matches = versionTagRegex(contents).exec(contents)
+  if (matches === null) {
+    throw new Error('Failed to read the <version> tag in your pom file - is it present?')
+  }
+
+  return matches[1]
+}
+
+module.exports.writeVersion = function (contents, version) {
+  //  Find the version tag, set the new version in it.
+  return contents.replace(versionTagRegex(contents), (match) => {
+    //  Replace the inner part of the version tag with the new version.
+    return match.replace(/>[^,]+</, `>${version}<`)
+  })
+}
```

Comment:
Parsing XML using regexps and indentation levels is a bad idea. If the intent with the chosen approach is to not add any unnecessary dependencies to the library, `xmldom` is pretty bare-bones and minimal, but would allow proper pom.xml parsing and serialization.

```suggestion
const { DOMParser, XMLSerializer } = require('xmldom')

function getDocAndVersionNode (contents) {
  const doc = new DOMParser().parseFromString(contents)
  const nodes = doc.documentElement.childNodes
  for (let i = 0; i < nodes.length; ++i) {
    const node = nodes[i]
    if (node.nodeName === 'version') return { doc, node }
  }
  return { doc }
}

module.exports.readVersion = function (contents) {
  const { node } = getDocAndVersionNode(contents)
  if (!node) {
    throw new Error('Failed to read the <version> tag in your pom file - is it present?')
  }
  const version = node.childNodes[0].data
  if (node.childNodes.length !== 1 || !version) {
    throw new Error('Expected <version> to contain a single string version')
  }
  return version
}

module.exports.writeVersion = function (contents, version) {
  const { doc, node } = getDocAndVersionNode(contents)
  node.childNodes[0].data = version
  return new XMLSerializer().serializeToString(doc)
}
```

Reply:
Hi, I know it has been already answered, but I tend to believe that if someone is working on a project with a pom.xml file, the java tooling (jdk/maven) should be present. Then a system call to maven is safer for version management, except that you are right about handling error and output of the sub-process.

Reply:
That would add a hard dependency on said tooling, which isn't actually always a reasonable assumption. For an example, I often need to work with JS packages that are deployed as WAR files. If I could use `standard-version` locally to set the pom.xml version, I could push that to version control and have a CI job do the actual release; at no step there would I necessarily need to locally use any Java tools.

All that's required here is using a proper XML parser/stringifier, as the position of the `<version>` in the structure is pretty well fixed.